content/en/guides/core/artifacts/track-external-files.md

```diff
@@ -94,76 +93,84 @@ artifact = run.use_artifact("mnist:latest", type="dataset")
-
-# Track the artifact and mark it as an input to
-# this run in one swoop. A new artifact version
-# is only logged if the files in the bucket changed.
-run.use_artifact(artifact)
+# Training here...
 
-artifact_dir = artifact.download()
+s3_client = boto3.client("s3")
+s3_client.upload_file(file_name="my_model.h5", bucket="my-bucket", object_name="models/cnn/my_model.h5")
 
-# Perform training here...
+# Log the model artifact
+model_artifact = wandb.Artifact("cnn", type="model")
+model_artifact.add_reference("s3://my-bucket/models/cnn/")
+run.log_artifact(model_artifact)
 ```
 
-To track models, we can log the model artifact after the training script uploads the model files to the bucket:
+At a later point, you can download the model artifact. Specify the name of the artifact and its type:
 
 ```python
-import boto3
 import wandb
 
 run = wandb.init()
-
-# Training here...
-
-s3_client = boto3.client("s3")
-s3_client.upload_file("my_model.h5", "my-bucket", "models/cnn/my_model.h5")
-
-model_artifact = wandb.Artifact("cnn", type="model")
-model_artifact.add_reference("s3://my-bucket/models/cnn/")
-run.log_artifact(model_artifact)
+artifact = run.use_artifact(artifact_or_name = "cnn", type="model")
+datadir = artifact.download()
 ```
 
 {{% alert %}}
-Read through the following reports for an end-to-end walkthrough of how to track artifacts by reference for GCP or Azure:
+See the following reports for an end-to-end walkthrough on how to track artifacts by reference for GCP or Azure:
 
 * [Guide to Tracking Artifacts by Reference](https://wandb.ai/stacey/artifacts/reports/Tracking-Artifacts-by-Reference--Vmlldzo1NDMwOTE)
 * [Working with Reference Artifacts in Microsoft Azure](https://wandb.ai/andrea0/azure-2023/reports/Efficiently-Harnessing-Microsoft-Azure-Blob-Storage-with-Weights-Biases--Vmlldzo0NDA2NDgw)
 {{% /alert %}}
 
-### Filesystem References
+## Cloud storage credentials
+
+W&B uses the default mechanism to look for credentials based on the cloud provider you use. Read the documentation from your cloud provider to learn more about the credentials used:
+
+| Cloud provider | Credentials Documentation |
+| -------------- | ------------------------- |
+| CoreWeave AI Object Storage (CAIOS)| [CAIOS documentation](https://docs.coreweave.com/docs/products/storage/object-storage/how-to/manage-access-keys) |
+| AWS            | [Boto3 documentation](https://boto3.amazonaws.com/v1/documentation/api/latest/guide/credentials.html#configuring-credentials) |
+| GCP            | [Google Cloud documentation](https://cloud.google.com/docs/authentication/provide-credentials-adc) |
+| Azure          | [Azure documentation](https://learn.microsoft.com/python/api/azure-identity/azure.identity.defaultazurecredential?view=azure-python) |
+
+For AWS, if the bucket is not located in the configured user's default region, you must set the `AWS_REGION` environment variable to match the bucket region.
+
+{{% alert color="secondary" %}}
```

Comment:
These points are somewhat orthogonal to each other as they cover entirely separate policies. What about giving the alert a good title that covers both, and making them two bullets?